### PR TITLE
add reference update scripts

### PR DESCRIPTION
**Pull request by lukasm91**
## Technical Description

Adds update scripts for all generated references in dawn and gtclang. 

All scripts try to guess the proper paths from yoda and normal build, otherwise, the options can be provided.

### Testing

Try to change some references and then run the update scripts.

### CODE REVIEW

File: dawn/test/unit-test/dawn/Optimizer/Passes/update_references.sh [hunk 1]
```diff
@@ -33,6 +33,10 @@ if [[ "$gtclang_path" == "" ]]; then
         echo "Cannot find gtclang."
         exit 1
     fi
+else
+    if [[ ! -f "$gtclang_path" ]]; then
```
Comment (twicki) [inline dawn/test/unit-test/dawn/Optimizer/Passes/update_references.sh]: how can we ever reach this?

Comment (lukasm91) [reply dawn/test/unit-test/dawn/Optimizer/Passes/update_references.sh]: if we are given an invalid path

Comment (twicki) [reply dawn/test/unit-test/dawn/Optimizer/Passes/update_references.sh]: :+1:

**Comment by lukasm91** [comment]
launch jenkins

**Pull request merged**